binding/ios/PorcupineManager.swift

```diff
@@ -11,150 +11,234 @@
+                outStatus.pointee = AVAudioConverterInputStatus.haveData
+                return buffer
+            }
+
+            converter?.convert(to: picoBuffer, error: nil, withInputFrom: input)
+            converter?.reset()
+
+            guard let pcm = picoBuffer.int16ChannelData?.pointee else {
+                return
+            }
+
+            var result: Int32 = -1
+
+            for i in 0..<Int(numberOfBuffer) {
+                let offsetPointer = pcm + i * Int(frameLength)
+                pv_porcupine_multiple_keywords_process(self.porcupine, offsetPointer, &result)
+                if result >= 0 {
+                    let index = Int(result)
+                    let keyword = self.wakeKeywordConfigurations[index]
+                    DispatchQueue.main.async {
+                        self.onDetection?(keyword)
+                    }
+                }
             }
         }
 
-        AudioQueueStart(queue, nil)
+        try audioEngine.start()
+    }
+
+    /// Stop listening for wake words.
+    public func stopListening() {
+
+        shouldBeListening = false
 
-        isListening = true
+        guard isListening else {
+            return
+        }
+
+        audioEngine.inputNode.removeTap(onBus: busNumber)
+        audioEngine.stop()
     }
 
-    /// Stops recording and releases resources acquired by Porcupine.
-    func stop() {
-        if !isListening { return }
-        
-        guard let queue = queue else { return }
-        AudioQueueStop(queue, true)
-        AudioQueueDispose(queue, false)
+    // MARK: - Private
 
-        pv_porcupine_delete(porcupine)
+    private func checkInitStatus(_ status: pv_status_t) throws {
+        switch status {
+        case PV_STATUS_IO_ERROR:
+            throw PorcupineManagerError.io
+        case PV_STATUS_OUT_OF_MEMORY:
+            throw PorcupineManagerError.outOfMemory
+        case PV_STATUS_INVALID_ARGUMENT:
+            throw PorcupineManagerError.invalidArgument
+        default:
+            return
+        }
+    }
 
-        isListening = false
+
+    @objc private func onAudioSessionInterruption(_ notification: Notification) {
+
+        guard let userInfo = notification.userInfo,
+            let typeValue = userInfo[AVAudioSessionInterruptionTypeKey] as? UInt,
+            let type = AVAudioSessionInterruptionType(rawValue: typeValue) else {
+                return
+        }
+
+        if type == .began {
+            // AVAudioEngine pauses on its own
+        } else if type == .ended {
+            // Interruption options are ignored. AVAudioEngine should be restarted
+            // unless PorcupineManager is told to stop listening.
+            guard let _ = userInfo[AVAudioSessionInterruptionOptionKey] as? UInt else {
+                return
+            }
+            if shouldBeListening {
+                try? audioEngine.start()
+            }
+        }
+    }
+}
+
+public struct WakeKeywordConfiguration {
```

Comment:
If possible let's change it to either WakeWordConfiguration or KeywordConfiguration. I think these two are a bit more known.